test/models/ptable_test.rb

```diff
@@ -70,11 +70,8 @@ class PtableTest < ActiveSupport::TestCase
   end
 
   test 'when creating a new ptable class object, an audit entry needs to be added' do
-    as_admin do
-      assert_difference('Audit.count') do
-        Ptable.create! :name => "dummy", :layout => "layout"
-      end
-    end
+    dummy_ptable = Ptable.create! :name => "dummy", :layout => "layout"
+    assert_equal 1, dummy_ptable.audits.where(action: 'create').count
```

Comment:
I am not so sure about this. Relative was better than absolute. Do we know why the test was randomly failing?

Reply:
@lzap,
With previous changes, it checks asset_difference for Audit.count not  `{ptable_instance}.audits.count`. So modified it with corresponding audits of Ptable record.

Reply:
Yeah, still relative seems to be better than absolute. I can have some audits in my test DB - leftovers and it will not be one. I am thinking like:

```ruby
before = Ptable.audits.where(action: 'create').count
dummy_ptable = Ptable.create! :name => "dummy", :layout => "layout"
assert_equal before + 1, dummy_ptable.audits.where(action: 'create').count
```

Reply:
There is no `audits` method available directly on model so I could get count using `auditable_type='Ptable'` as below.

```suggestion   
    before =  Audit.where(auditable_type: 'Ptable', action: 'create').count
    dummy_ptable = Ptable.create! :name => "dummy", :layout => "layout"
    assert_equal before + 1, Audit.where(auditable_type: 'Ptable', action: 'create').count
```

Still one query - what difference it will make using before-after count? Instead it would be good to verify audits count directly for that instance than audits count from model level.

Reply:
You suggestion looks better to me than absolute count check.



Not following what you actually have on your mind 

> Instead it would be good to verify audits count directly for that instance than audits count from model level.

Not following.

Reply:
TBH i don't know if we really need this test. This is essentially testing that the audited gem works as expected, which should be tested by audited and not by us. or am I missing something here?

Reply:
I agree, actually we have *five* (?) of them. Maybe let's just delete these? Amount of burden we've had with random failures was just enough.

```
[lzap@box foreman]$ ag "assert_difference..Audit"
test/models/lookup_key_test.rb
235:    assert_difference('Audit.count') do

test/models/ptable_test.rb
74:      assert_difference('Audit.count') do

test/models/lookup_value_test.rb
138:    assert_difference('Audit.count') do
149:    assert_difference('Audit.count') do

test/models/hostgroup_class_test.rb
13:      assert_difference('Audit.count') do
```

Reply:
@tbrisker, @lzap,
should I delete all of those?

Reply:
If all they are testing is that audited works (like this one), i vote to delete.

Reply:
Or - if we want to keep a test to make sure the class is audited, we can do that without generating an actual audit.

Reply:
I agree, just check if they are really just dumb audit tests and let them go.

Reply:
Surely! As you both are agree, I am going to remove them. 
Apart from checking an audit record creation, they are not testing anything else.